### PR TITLE
Adds the ability to enable basic observations per service

### PR DESCRIPTION
**Pull request by chrisbsmith**
**Please provide enough information so that others can review your pull request:**

1. Explain the **details** for making this change. What existing problem does the pull request solve? If it resolves an existing issue, be sure to use a Github [keyword](https://help.github.com/en/articles/closing-issues-using-keywords) to automatically close it.

This PR adds the ability to enable or disable observables per service.  

2. What **changes to custom.yaml** are required?

This new section is needs to be added to `.Values.global`:

```

3. Have you documented any additional setup steps or configurations options?

<br/><br/>

4. Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.

<br/><br/>

### CODE REVIEW

File: docs/Getting Started.md [hunk 1]
```diff
@@ -91,6 +92,21 @@ Once the edge proxy is deployed, voyager-operator will create a custom ingress r
...
+
+The Grey Matter Helm chart provides the ability to enable observables in the mesh, but they are disabled by default.  To enable observables, modify the following settings in your local custom values file.  The `kafkaServerConnection` should be a reference to a kafka ensemble that is available in the mesh.  The format is `<host1>:<port1>,<host2>:<port2>`
+
+These are global settings for all observables:
+
+```yaml
+globals:
+  observables:
+    topic: observables
+    kafkaServerConnection:
```
Comment (tilleryd) [inline docs/Getting Started.md]: I think we should provide valid YAML. A key without a value throws errors with older versions of Helm (despite we don't support them). Also it's just good practice.

File: greymatter/values.yaml [hunk 2]
```diff
@@ -4,6 +4,12 @@ global:
   route_url_name:
   remove_namespace_from_url: false
   zone: zone-default-zone
+  # The observables section defines the common information points for
+  # Grey Matter Observables.  Note that, if enabled, the kafkaServerConnection
+  # MUST be enabled
+  observables:
+    topic: observables
+    kafkaServerConnection:
```
Comment (tilleryd) [inline greymatter/values.yaml]: Again, I think this key should have a value.

File: docs/Getting Started.md (hunk 1 above)
Comment (chrisbsmith) [reply docs/Getting Started.md]: So I thought about this, but if I provide something like `localhost:9092` and the user enables it without changing this value, there are more severe consequences in the mesh.  Open to it if you really thing we should.

File: greymatter/values.yaml [hunk 3]
```diff
@@ -68,6 +74,7 @@ global:
       catalogEnabled: true
       controlEnabled: true
       serviceName: 'catalog'
+      observables: false
```
Comment (tilleryd) [inline greymatter/values.yaml]: We should be consistent with boolean naming conventions. e.g. `observablesEnabled`.

File: gm-control-api/json/services/proxy.json [hunk 4]
```diff
@@ -9,9 +9,23 @@
...
+    {{-  if (.service.observables) }}
+    "gm_observables": {
+      "useKafka": true,
+      "topic": "{{.service.serviceName}}",
+      "eventTopic": "{{ .Values.global.observables.topic }}",
+      "kafkaServerConnection": "{{ .Values.global.observables.kafkaServerConnection }}"
+    }
+    {{- end }}
     "gm_metrics": {
+      "service_observable": "{{.service.observables}}"
```
Comment (tilleryd) [inline gm-control-api/json/services/proxy.json]: I'm not aware of this as a k/v pair used by metrics. What does it do?

File: greymatter/values.yaml (hunk 3 above)
Comment (chrisbsmith) [reply greymatter/values.yaml]: 👍

File: docs/Getting Started.md [hunk 5]
```diff
@@ -91,6 +92,21 @@ Once the edge proxy is deployed, voyager-operator will create a custom ingress r
 
 Read [Ingress](./Ingress.md) for further details.
 
+### Observables
+
+The Grey Matter Helm chart provides the ability to enable observables in the mesh, but they are disabled by default.  To enable observables, modify the following settings in your local custom values file.  The `kafkaServerConnection` should be a reference to a kafka ensemble that is available in the mesh.  The format is `<host1>:<port1>,<host2>:<port2>`
```
Comment (tilleryd) [inline docs/Getting Started.md]: `Kafka` (uppercase)

Comment (tilleryd) [reply docs/Getting Started.md]: I updated this doc to instruct the user to create a `custom-greymatter.yaml` file. I suggest we have them add this config there. i.e. `modify the following settings in custom-greymatter.yaml.`.

Comment (chrisbsmith) [reply docs/Getting Started.md]: done.

File: gm-control-api/json/services/proxy.json (hunk 4 above)
Comment (chrisbsmith) [reply gm-control-api/json/services/proxy.json]: Good catch, that was cruft I forgot to remove.

File: docs/Getting Started.md (hunk 1 above)
Comment (tilleryd) [reply docs/Getting Started.md]: Hmm, now that I think about it, there isn't a value to add until they've setup Kafka. So I guess this is adequate.

File: greymatter/values.yaml (hunk 2 above)
Comment (tilleryd) [reply greymatter/values.yaml]: Never mind, same realization as my comment on the Getting Started doc.

**Pull request merged**